lib/Doctrine/DBAL/Portability/Connection.php

```diff
@@ -66,7 +67,7 @@ public function connect()
             }
 
             if (isset($params['fetch_case']) && $this->portability & self::PORTABILITY_FIX_CASE) {
-                if ($this->_conn instanceof PDOConnection) {
+                if ($this->_conn instanceof PDOConnection || $this->_conn instanceof WrappedPDOConnection) {
```

Comment:
Is there a chance that `$this->_conn` can be of one type only?

Reply:
Considering that modifying `PDOConnection` to use wrapped instances would be a BC-breaking change, I'd say no =/

I just modified things here too to be 100% safe that things work in a similar way to when we create the instance...